posts/Yoann/2017-11-27-introducing-hibernate-search-jsr352-mass-indexing-job.adoc

```diff
@@ -0,0 +1,163 @@
+We are in the process of testing at least one other JEE application server and Spring Batch:
+see <<limitations-and-next>> below.
+
+=== Usage
+
+All the examples below assume the variable `jobOperator` contains
+the job operator retrieved as instructed above.
+
+To launch a mass indexing job:
+
+```
+java.util.Properties parameters = org.hibernate.search.jsr352.massindexing.MassIndexingJob.parameters()
+		.forEntities( MyFirstEntity.class, MySecondEntity.class )
+		.build();
+long executionId = jobOperator.start(
+		org.hibernate.search.jsr352.massindexing.MassIndexingJob.NAME,
+		parameters
+		);
+```
+
+You can then query the execution status through this call:
+
+```
+javax.batch.runtime.JobExecution jobExecution = jobOperator.getJobExecution( executionId );
+```
+
+You can suspend the job this way:
+
+```
+long executionId2 = jobOperator.stop( executionId );
+```
+
+To resume the job after suspension or upon failure, you can request a restart this way:
+
+```
+long executionId2 = jobOperator.restart( executionId, parameters );
+```
+
+[TIP]
+====
+According to the JSR-352 specification, you must provide the parameters again when restarting a job.
+
+However, https://github.com/jberet/jsr352#readme[JBeret], the JSR-352 implementation included in Wildfly,
+allows you to pass `null` instead, and will interpret it as
+"the parameters I used when initially starting the job".
+====
+
+The builder you get from calling `MassIndexingJob.parameters()` offers a few settings.
+Here are some of the most useful:
+
+* `.purgeAllOnStart(boolean)`: specify whether the existing index should be purged at the beginning of the job.
+* `.restrictedBy(Criterion)`: use a `Criterion` to only index a subset of entities.
+* `.restrictedBy(String)`: use HQL to only index a subset of entities.
+  Note that when using HQL, the job switches to a degraded mode that suffers from several limitations,
+  making it fit only for small, quick executions.
+  See https://docs.jboss.org/hibernate/search/5.9/reference/en-US/html_single/#jsr-352-indexing-mode[the documentation] for details.
+
+You can find more information about the job parameters, how to tune performance,
+or how to deal with applications relying on multiple JPA persistence units
+in https://docs.jboss.org/hibernate/search/5.9/reference/en-US/html_single/#jsr352-integration[the documentation].
+
+== Performance
+
+We did a bit of benchmarking, and as far as we could see,
+the performance of the JSR-352 mass indexing job is on par
+with that of the existing mass indexer.
+
+However, please keep in mind that performance depends a lot on your setup,
+and you may have to tinker a bit with the job parameters in order to achieve optimal performance,
+as explained in https://docs.jboss.org/hibernate/search/5.9/reference/en-US/html_single/#jsr352-integration[the documentation].
+
+== Credits
+
+I would like to mention that a large part of this JSR-352 integration was developed
```

Comment:
I think this should be mentioned right at the beginning (E.g. "Originally started as a GSoC project by Mincong Huang, we're today happy to announce the release of ...").